com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/Bob.java

```diff
@@ -334,7 +335,34 @@ public static String getJarFile(String filename) throws IOException {
         return f.getAbsolutePath();
     }
 
-    public static String getDefaultDmengineExeName(String variant) {
+    private static List<File> downloadExes(Platform platform, String variant) throws IOException {
+        init();
+        TimeProfiler.startF("DownloadExes %s for %s", platform, variant);
+        List<File> binaryFiles = new ArrayList<File>();
+        String[] exeSuffixes = platform.getExeSuffixes();
+        List<String> exes = new ArrayList<String>();
+        String downloadFolder = rootFolder + File.pathSeparator + platform.getPair() + File.separator + platform;
+        String defaultDmengineExeName = getDefaultDmengineExeName(variant);
+        for (String exeSuffix : exeSuffixes) {
+            String exeName = platform.getExePrefix() + defaultDmengineExeName + exeSuffix;
+            File f = new File(rootFolder, exeName);
+            try {
+                URL url = new URL(String.format("http://d.defold.com/archive/%s/engine/%s/%s", EngineVersion.sha1, platform.getPair(), exeName));
```

Comment:
What about private platforms (consoles)? Unlikely that there is a need for headless versions of the engine though, but I don't think we should hardcode d.defold.com. The files may be located elsewhere.